python/cuml/metrics/cluster/silhouette_score.py

```diff
@@ -0,0 +1,72 @@
+import cupy as cp
+
+from cuml.metrics.cluster.silhouette_score_impl import _silhouette_coeff
+
+
+def silhouette_score(
+        X, labels, metric='euclidean', handle=None):
+    """Calculate the mean silhouette coefficient for the provided data
+
+    Given a set of cluster labels for every sample in the provided data,
+    compute the mean intra-cluster distance (a) and the mean nearest-cluster
+    distance (b) for each sample. The silhouette coefficient for a sample is
+    then (b - a) / max(a, b).
+
+    Parameters
+    ----------
+    X : array-like, shape = (n_samples, n_features)
+        The feature vectors for all samples.
+    labels : array-like, shape = (n_samples,)
+        The assigned cluster labels for each sample.
+    metric : string
+        A string representation of the distance metric to use for evaluating
+        the silhouette schore. Available options are "cityblock", "cosine",
+        "euclidean", "l1", "l2", "manhattan", and "sqeuclidean".
+    handle : cuml.Handle
+        Specifies the cuml.handle that holds internal CUDA state for
+        computations in this model. Most importantly, this specifies the CUDA
+        stream that will be used for the model's computations, so users can
+        run different models concurrently in different streams by creating
+        handles in several streams.
+        If it is None, a new one is created.
+    """
+
+    return _silhouette_coeff(
+        X, labels, metric=metric, handle=handle
+    )
+
+
+def silhouette_samples(X, labels, metric='euclidean', handle=None):
```

Comment:
Can't this be done in the same `silhouette_score` function above? Or is this following an existing (scikit-learn or somewhere else) API?

Reply:
This is to follow the sklearn API.

- https://scikit-learn.org/stable/modules/generated/sklearn.metrics.silhouette_score.html
- https://scikit-learn.org/stable/modules/generated/sklearn.metrics.silhouette_samples.html#sklearn.metrics.silhouette_samples